### PR TITLE
[Behavioral analytics] Implement search filters into events.

### PR DESCRIPTION
**Pull request by afoucret**
Now supporting filters into search and search_click events:

Example payload:

```json
{
  "search": {
    "query": "air jordan", 
    "filters": { 
      "brand": "nike", 
      "categories": ["shoes", "sales"]
    },
    "search_application": "website"
  },
  "user": {
    "id": "d682f5af-c73e-46c3-bae7-14f11e191178"
  },
  "session": {
    "id": "ae045ac4-8a37-4b60-9125-0610227b27fd"
  }
}
```

### CODE REVIEW

File: x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/analytics/event/parser/field/SearchFiltersAnalyticsEventFieldTests.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.Mockito.mock;
+
+public class SearchFiltersAnalyticsEventFieldTests extends AnalyticsEventFieldParserTestCase<List<String>> {
+    @Override
+    public void testParsingWithInvalidField() {
+        // No invalid field since we are parsing a map.
+    }
+
+    public void testParsingInvalidFiltersValues() throws IOException {
```
Comment (joemcelroy) [inline x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/analytics/event/parser/field/SearchFiltersAnalyticsEventFieldTests.java:34]: nice!

**Review by joemcelroy — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/entsearch/analytics/behavioral_analytics-events-mappings.json [hunk 2]
```diff
@@ -86,16 +86,7 @@
...
-                "field": {
-                    "type": "keyword",
-                    "ignore_above": 1024
-                },
-                "value": {
-                    "type": "keyword",
-                    "ignore_above": 1024
-                }
-              }
+              "type": "flattened"
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/entsearch/analytics/behavioral_analytics-events-mappings.json:89]: Looks fine to use `flattened` for now but that reminds me that we should discuss the mapping a bit more. We have a lot of `ignore_above` on keyword fields so I wonder what's the expectations on these fields.

**Review by jimczi — APPROVED** [review]
LGTM
Let's review the full mapping in a follow up.

**Pull request merged**